SteamKit2/SteamKit2/Steam/CDN/LancacheDetector.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * This file is subject to the terms and conditions defined in
+ * file 'license.txt', which is part of this source code package.
+ */
+
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+
+namespace SteamKit2.Steam.CDN
+{
+    /// <summary>
+    /// Attempts to automatically resolve the Lancache's IP address in the same manner that the Steam client does.
+    ///
+    /// Will automatically try to detect the Lancache through the poisoned DNS entries.
+    /// This is a modified version from the original source : https://github.com/tpill90/lancache-prefill-common/blob/main/dotnet/LancacheIpResolver.cs
+    /// </summary>
+    public static class LancacheDetector
```

Comment:
Make `CDN.Client` partial, and then extend it here, then you can put all the lancache code in the same file.

Reply:
@xPaw I finally got back to looking at this.  Broke the lancache specific code out into this file, and made it an explicit flag that needs to be enabled in order to use the lancache.  Can you take another look and check if the current code is closer to what you were thinking?